typing/typecore.ml

```diff
@@ -40,6 +40,18 @@ type type_expected = {
   explanation: type_forcing_context option;
 }
 
+module Label_kind = struct
```

Comment:
As we discussed, labels are field names, not constructor names.
So this should probably be `Name_kind` or `Choice_kind` ?

By the way, does it really need to be a module?

Reply:
It is a module because the auxiliary functions are too hard to name properly otherwise -- we know, we tried without a module first.

Reply:
Re. `label` rather than `name` or `choice`: the code had a `label_of_kind` function before this PR which returned either `"field"` or `"constructor"`, so I think that using "label" for "either a field name or a record name" is not too wrong -- not worse than before, and not worse that the two naming choices you propose, which are too generic and unclear.

Reply:
As I repeat, the reason it was called `label_of_kind` was because the functor was obtained by generalizing code specialized for field labels. But here you define a new type independent of the functor. Please think of a correct name.

Reply:
What about `Description_kind`, then.

Reply:
What about `Description_kind`, then.

Reply:
FTR I agree that Label is the wrong choice for this.

Reply:
I personally don't see anything wrong with using `label` to mean "a variant constructor or a record field (or a method name or a polymorphic variant constructor or an argument label)". I don't mind using a different name if you have one to suggest, but I don't think "name" or "choice" are any good. "Choice" doesn't mean anything in the context and "Name" could be just any identifier. Label may not be ideal, but again, the existing code did use `label` all over the place; if it was good enough then, maybe it's good enough now?

Reply:
After discussing with @garrigue,  a possible name  would be `member`: the module could be named `Member_kind` and the `label_name` function would become `member_name`. (Other propositions included `Datatype_kind`, `Datatype_member`.)